artifactory_cleanup/rules/filters.py

```diff
@@ -68,3 +68,18 @@ class ExcludeFilename(FilterRule):
     attribute_name = "name"
     operator = "$nmatch"
     boolean_operator = "$and"
+
+
+class FilterByRegexpPath(Rule):
+    """
+    Filter artifacts paths by regex pattern.
+    """
+
+    def init(self, path):
+        self.path = rf"{path}"
+
+    def filter(self, artifacts: ArtifactsList) -> ArtifactsList:
+        for artifact in artifacts[:]:
+            if re.match(self.path, artifact["path"]) is None:
+                artifacts.remove(artifact)
```

Comment:
One more for readabilty
```suggestion
                artifacts.keep(artifact)
```